csaf/src/source/file.rs

```diff
@@ -108,12 +108,17 @@ impl FileSource {
     ) -> Result<mpsc::Receiver<walkdir::Result<walkdir::DirEntry>>, anyhow::Error> {
         let (tx, rx) = mpsc::channel(8);
 
-        let path = distribution.directory_url.to_file_path().map_err(|()| {
-            anyhow!(
-                "Failed to convert into path: {}",
-                distribution.directory_url
-            )
-        })?;
+        let path = distribution
+            .directory_url
+            .clone()
+            .unwrap()
```

Comment:
Unwrapping might panic which feels wrong in this location. Please find a way to properly handle the case.